02-hello-accessibility/projects/starter/app/src/main/java/com/raywenderlich/android/tacotuesday/data/DataFactory.kt

```diff
@@ -0,0 +1,2262 @@
+                "* [Tomatoes](../mixins/tomatoes_traditional_us.md)\n" +
+                "* [Jalapeños](../mixins/jalapenos.md)\n" +
+                "* [Cilantro](../mixins/cilantro.md)\n" +
+                "\n" +
+                "Divide the cooked Mac n Cheese up into several bowls. Top each serving with the " +
+                "ground beef, sour cream, tomatoes, jalapeños, and cilantro. Delish. Feel free to get extra fancy and add extra taco toppings that you enjoy.",
+            baseLayer = groundBeef,
+            condiment = sourCream + "\n" + tomatoes + "\n" +
+                "Jalapeños\n" +
+                "==========\n" +
+                "\n" +
+                "What are tacos without a bit of spice? These deliciously fiery peppers have earned their spot as a staple in Mexican food preparations. Chances are, they'll make a great addition to your taco recipe.\n" +
+                "\n" +
+                "* Jalapeños (sliced)\n" +
+                "\n" +
+                "tags: vegetarian\n" +
+                "Cilantro\n" +
+                "==========\n" +
+                "\n" +
+                "Sometimes when you're eating a taco, you think to yourself \"this is missing something, and I can't figure out what it is.\" I'm willing to bet that something is cilantro. This subtle yet oh so important flavor enhancer will take your tacos to the next level.\n" +
+                "\n" +
+                "* Cilantro (chopped)\n" +
+                "\n" +
+                "tags: vegetarian\n",
+            photoRes = R.drawable.taco_3),
+        Recipe(
+            name = "Taco Soup",
+            recipe = "Taco Soup\n" +
+                "===================\n" +
+                "Approximately 12 servings\n" +
+                "\n" +
+                "* 2 pounds ground beef\n" +
+                "* 2 cup diced onions\n" +
+                "* 2 (14 1/2-ounce) cans Mexican-style stewed tomatoes\n" +
+                "* 1 (14 1/2-ounce) can diced tomatoes\n" +
+                "* 1 (15 1/4-ounce) can whole kernel corn, drained\n" +
+                "* 2 (15 1/2-ounce) cans pinto beans\n" +
+                "* 1 (15 1/2-ounce) can kidney beans\n" +
+                "* 2 (4 1/2-ounce) cans diced green chiles\n" +
+                "* 1 (4.6-ounce) can black olives, drained & sliced\n" +
+                "* 1 (1 1/4-ounce) package taco seasoning mix\n" +
+                "* 1 (1-ounce) package ranch salad dressing mix\n" +
+                "* Sour cream\n" +
+                "* Grated cheese\n" +
+                "* Corn chips\n" +
+                "\n" +
+                "1. Brown ground beef and onions, and drain fat\n" +
+                "2. Combine gound beef, onions and other ingredients (except sour cream, grated cheese and corn chips) in slow cooker\n" +
+                "3. Cook for 6-8 hours \n" +
+                "4. Serve with sour cream, grated cheese and corn chips",
+            photoRes = R.drawable.taco_4)
```

Comment:
On the one hand, having this in Kotlin rather than in a file or files seems like it'd be difficult to maintain long term. On the other hand, I don't know what we'd need to edit longer-term. What's your thought in doing this in code @vgonda?

Reply:
Lol my thought for this file is "shove it all in a factory so we don't need to rely on an external API" 😂 This is easier to keep reliable than someone else's API. I expect this to never change, or not change much. I have no plans to touch this in the book.